src/Icicle/Sea/FromAvalanche/Analysis.hs

```diff
@@ -129,20 +133,21 @@ readsOfProgram = readsOfStatement . statements
 readsOfStatement :: Eq n => Statement (Annot a) n Prim -> Map (Name n) (ValType)
 readsOfStatement stmt
  = case stmt of
-     Block []              -> Map.empty
-     Block (s:ss)          -> readsOfStatement s `Map.union`
-                              readsOfStatement (Block ss)
-     Let _ _ ss            -> readsOfStatement ss
-     If _ tt ee            -> readsOfStatement tt `Map.union`
-                              readsOfStatement ee
-     ForeachInts  _ _ _ ss -> readsOfStatement ss
-     ForeachFacts _ _ _ ss -> readsOfStatement ss
-     InitAccumulator _ ss  -> readsOfStatement ss
-     Write _ _             -> Map.empty
-     LoadResumable _ _     -> Map.empty
-     SaveResumable _ _     -> Map.empty
-     Output _ _ _          -> Map.empty
-     KeepFactInHistory _   -> Map.empty
+     Block []                -> Map.empty
+     Block (s:ss)            -> readsOfStatement s `Map.union`
+                                readsOfStatement (Block ss)
+     Let _ _ ss              -> readsOfStatement ss
+     If _ tt ee              -> readsOfStatement tt `Map.union`
+                                readsOfStatement ee
+     While          _ _ _ ss -> readsOfStatement ss
```

Comment:
in some sense this is actually reading from the accumulator every time. I am not sure what `readsOfStatement` is actually used for though, so it might not be necessary to add it here